modules/grizzly/src/test/java/org/glassfish/grizzly/FilterChainReadTest.java

```diff
@@ -16,45 +16,58 @@
 
 package org.glassfish.grizzly;
 
+import java.io.EOFException;
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.security.SecureRandom;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedTransferQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 import org.glassfish.grizzly.asyncqueue.WritableMessage;
 import org.glassfish.grizzly.filterchain.BaseFilter;
 import org.glassfish.grizzly.filterchain.FilterChain;
 import org.glassfish.grizzly.filterchain.FilterChainBuilder;
 import org.glassfish.grizzly.filterchain.FilterChainContext;
 import org.glassfish.grizzly.filterchain.NextAction;
 import org.glassfish.grizzly.filterchain.TransportFilter;
+import org.glassfish.grizzly.memory.CompositeBuffer;
+import org.glassfish.grizzly.nio.transport.TCPNIOConnectorHandler;
 import org.glassfish.grizzly.nio.transport.TCPNIOTransport;
 import org.glassfish.grizzly.nio.transport.TCPNIOTransportBuilder;
 import org.glassfish.grizzly.utils.EchoFilter;
 import org.glassfish.grizzly.utils.StringEncoder;
 import org.glassfish.grizzly.utils.StringFilter;
-import java.io.EOFException;
-import java.io.IOException;
-import java.net.SocketAddress;
-import java.util.concurrent.BlockingQueue;
-import java.util.concurrent.Future;
-import java.util.concurrent.LinkedTransferQueue;
-import java.util.concurrent.TimeUnit;
-import java.util.logging.Level;
-import java.util.logging.Logger;
+
 import junit.framework.TestCase;
-import org.glassfish.grizzly.memory.CompositeBuffer;
-import org.glassfish.grizzly.nio.transport.TCPNIOConnectorHandler;
 
 /**
  * Test {@link FilterChain} blocking read.
- * 
+ *
  * @author Alexey Stashok
  */
 @SuppressWarnings("unchecked")
 public class FilterChainReadTest extends TestCase {
-    public static final int PORT = 7785;
+    public static final int PORT = PORT();
+
+    static int PORT() {
```

Comment:
When I saw this is left with default visibility I hoped it is reused. Unfortunately I see the same code duplicated several times instead. Why not make initial port base argument of this method (not the best name IMO, as there is already field of that name, and I'd like the method to have a verb) and have only one?

Reply:
I had the idea of calling the PORT() method from within the code whenever the "cannot bind to port exception occurs". But actually the exception is often not "true", and it's just a matter of a sleep. For now, it mostly needed a quick way to pass the tests and do away with stuck ports.

Better would be to move the port code, AND the connection code, to a test utility class and have all tests use that. That way the get port, sleep, retry and obtain new port, can be kept at one place.

Reply:
p.s. there's quite a bit more cleaning to do for Grizzly. Specifically the `Connection` type is missing generics everywhere and there's a bunch of JavaDoc warnings when building, as well as forkCount warnings etc.

Reply:
Also many warnings about deprecated types:

```
[WARNING] /home/travis/build/eclipse-ee4j/grizzly/modules/http2/src/main/java/org/glassfish/grizzly/http2/Http2Stream.java:[41,29] org.glassfish.grizzly.ICloseType in org.glassfish.grizzly has been deprecated
```